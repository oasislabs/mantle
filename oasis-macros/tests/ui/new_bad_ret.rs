oasis_std::contract! {

#[derive(Contract)]
#[derive(Default)]
pub struct Counter(u32);

impl Counter {
    pub fn new(ctx: Context, start_count: u32) -> () {
        Self(start_count);
    }
}

}

fn main() {}
