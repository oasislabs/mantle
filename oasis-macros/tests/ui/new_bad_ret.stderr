error: `Counter::new` must return `Self`
 --> $DIR/new_bad_ret.rs:8:49
  |
8 |     pub fn new(ctx: &Context, start_count: u32) -> () {
  |                                                 ^^

error: aborting due to previous error

